src/transform/resize.rs

```diff
@@ -128,11 +126,42 @@ impl WriteTags for Resize {
     }
 }
 
+impl<T> TryFrom<&mut Decoder<T>> for Resize
+where
+    T: Read + Seek,
+{
+    type Error = ResizeError;
+
+    /// Read the resize metadata from a TIFF file
+    fn try_from(decoder: &mut Decoder<T>) -> Result<Self, Self::Error> {
+        let scale = decoder
+            .get_tag_f32_vec(Tag::Unknown(DEFAULT_SCALE))
+            .context(ReadTiffTagSnafu {
+                name: "DefaultScale",
+            })?;
+        if scale.len() != 2 {
```

Comment:
I didn't like the idea of using full names like "scale_x" which aren't otherwise used, but I agree that the magic number here doesnt look good. I added a constant to the respective structs indicating the expected number of items from each tag. I also added some `From` implementations to help converting the structs to tuples